qtp_job_output_folder/validate.py

```diff
@@ -0,0 +1,67 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from os.path import exists, isdir
+
+from json import loads
+
+from .summary import _generate_html_summary
+
+from qiita_client import ArtifactInfo
+
+
+def validate(qclient, job_id, parameters, out_dir):
+    """Validae and fix a new artifact
+
+    Parameters
+    ----------
+    qclient : qiita_client.QiitaClient
+        The Qiita server client
+    job_id : str
+        The job id
+    parameters : dict
+        The parameter values to validate and create the artifact
+    out_dir : str
+        The path to the job's output directory
+
+    Returns
+    -------
+    dict
+        The results of the job
+
+    Raises
+    ------
+    ValueError
+        If there is any error gathering the information from the server
+    """
+    qclient.update_job_step(job_id, "Step 1: Collecting prep information")
+
+    files = loads(parameters['files'])
+    # [0] we only expect one directory
+    folder = files['directory'][0]
+
+    success = False
+    ainfo = None
+    error_msg = f'{folder} does not exist or is not a folder'
+    if exists(folder) and isdir(folder):
```

Comment:
Should we check that we have reading permissions?

Reply:
We could but I'm not sure is needed.

Reply:
Sounds good!